Xamarin.Essentials/Contacts/Contacts.ios.cs

```diff
@@ -27,6 +29,46 @@ static Task<Contact> PlatformPickContactAsync()
             return source.Task;
         }
 
+        static async Task<IEnumerable<Contact>> PlatformGetAllAsync()
+        {
+            await Task.CompletedTask;
+            return GetAll();
+        }
+
+        static IEnumerable<Contact> GetAll()
+        {
+            var keys = new[]
+            {
+                CNContactKey.NamePrefix,
+                CNContactKey.GivenName,
+                CNContactKey.MiddleName,
+                CNContactKey.FamilyName,
+                CNContactKey.NameSuffix,
+                CNContactKey.EmailAddresses,
+                CNContactKey.PhoneNumbers
+            };
+
+            var results = new List<Contact>();
+            using (var store = new CNContactStore())
+            {
+                var containers = store.GetContainers(null, out var error);
+
+                foreach (var container in containers)
+                {
+                    try
+                    {
+                        using var pred = CNContact.GetPredicateForContactsInContainer(container.Identifier);
+                        results.AddRange(store.GetUnifiedContacts(pred, keys, out error)?.Select(a => GetContact(a)));
```

Comment:
We need to see if this is possible on iOS, UWP, MacOS and Tizen.

Reply:
@mattleibow 
First we need to write a draft implementation for all platforms

Reply:
Of course. Just wondering if it will even be possible... I know UWP likes to have awaits everywhere. That one might be the real one to cause any issues...

Reply:
There are TasCompletionSource for emergencies